app/src/main/java/com/hedvig/app/feature/keygear/KeyGearValuationInfoActivity.kt

```diff
@@ -0,0 +1,67 @@
+package com.hedvig.app.feature.keygear
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import com.hedvig.android.owldroid.fragment.KeyGearItemFragment
+import com.hedvig.app.BaseActivity
+import com.hedvig.app.R
+import com.hedvig.app.util.extensions.observe
+import com.hedvig.app.util.extensions.setMarkdownText
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import com.hedvig.app.util.interpolateTextKey
+import kotlinx.android.synthetic.main.activity_key_gear_valuation_info.*
+import org.koin.android.viewmodel.ext.android.viewModel
+
+class KeyGearValuationInfoActivity : BaseActivity(R.layout.activity_key_gear_valuation_info) {
+
+    private val model: KeyGearValuationInfoViewModel by viewModel()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val id = intent.getStringExtra(ITEM_ID)
+
+        close.setHapticClickListener {
+            onBackPressed()
+        }
+
+        model.data.observe(this) { data ->
+            data?.let {
+                val category = data.fragments.keyGearItemFragment.category.toString().toLowerCase()
+                val purchasePrice = data.fragments.keyGearItemFragment.purchasePrice?.amount
+                val valuationPercentage =
+                    (data.fragments.keyGearItemFragment.valuation as KeyGearItemFragment.AsKeyGearItemValuationFixed).ratio
```

Comment:
This cast is highly unsafe. We need to behave differently based on the type here.

Use a `when`-statement:

```kotlin
when (val valuation = data.fragments.keyGearItemFragment.valuation)
    is KeyGearItemFragment.AsKeyGearItemValuationFixed -> { ... }
    is KeyGearItemFragment.AsKeyGearItemValuationMarketPrice -> { ... }
    else -> { ... }
```

Reply:
Come to think of it - this screen has no interaction whatever with the data except for displaying it.

We could actually make it entirely parameterized and avoid using a ViewModel entirely - this would probably be quite desirable actually as it decreases complexity significantly 🙂